master/internal/db/postgres_groups.go

```diff
@@ -0,0 +1,194 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pkg/errors"
+
+	"github.com/jackc/pgconn"
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+// AddGroup adds a group to the database. Returns ErrDuplicateRow if a
+// group already exists with the same name or ID.
+func (db *PgDB) AddGroup(ctx context.Context, group model.Group) (model.Group, error) {
+	_, err := Bun().NewInsert().Model(&group).Exec(ctx)
+	return group, matchSentinelError(err)
+}
+
+// GroupByID looks for a group by id. Returns ErrNotFound if the group isn't found.
+func (db *PgDB) GroupByID(ctx context.Context, gid int) (model.Group, error) {
+	var g model.Group
+	err := Bun().NewSelect().Model(&g).Where("id = ?", gid).Scan(ctx)
+
+	return g, matchSentinelError(err)
+}
+
+// SearchGroups searches the database for groups. userBelongsTo is "optional"
+// in that if a value < 1 is passed in, the parameter is ignored. SearchGroups
+// does not return an error if no groups are found, as that is considered a
+// successful search.
+func (db *PgDB) SearchGroups(ctx context.Context,
+	userBelongsTo model.UserID) ([]model.Group, error) {
```

Comment:
If that's a strong preference, that seems okay with me.

I tend to like to avoid extraneous pointers when I can, and I think calling it with a non-pointer signature is a little less of a hassle. For example:

```go
g, err := db.SearchGroups(ctx, 12)
g, err := db.SearchGroups(ctx, user.ID)
g, err := db.SearchGroups(ctx, 0)
```
vs
```go
var id model.UserID = 12
g, err := db.SearchGroups(ctx, &id)
g, err := db.SearchGroups(ctx, &user.ID)
g, err := db.SearchGroups(ctx, nil)
```

Though I totally recognize the oddness of passing in an out-of-band value as a sentinel to ignore a parameter.

Reply:
I also somewhat prefer nil pointers over special errors